src/commands/logs.ts

```diff
@@ -0,0 +1,323 @@
+        let logPage = 0;
+        let selectedCategory = Logs.User;
+
+        const replyOptions = async (): Promise<InteractionReplyOptions> => ({
+            embeds: [
+                new EmbedBuilder()
+                    .setTitle(`${member.displayName}'s moderation history`)
+                    .setDescription(
+                        await this.renderLogPage(
+                            selectedCategory,
+                            logPage,
+                            userLog,
+                            cachedPages
+                        )
+                    ),
+            ],
+            components: [
+                await this.selectActionRow(selectedCategory),
+                await this.buttonActionRow(logPage, await this.getLogLength(selectedCategory, userLog)),
+            ],
+        });
+        const reply = await interaction.editReply({
+            fetchReply: true,
+            ...(await replyOptions()),
+        });
+
+        const collector = reply.createMessageComponentCollector();
+        collector.on('collect', async i => {
+            await i.deferUpdate();
+            if (i.customId === Logs.ButtonPrev) {
+                logPage--;
+            } else if (i.customId === Logs.ButtonNext) {
+                logPage++;
+            } else if (i.isStringSelectMenu() && i.customId === Logs.Category) {
+                selectedCategory = i.values[0] as Logs;
+                logPage = 0;
+            }
+            await i.editReply(await replyOptions());
+        });
+    }
+
+    public override registerApplicationCommands(registry: Command.Registry) {
+        registry.registerChatInputCommand(
+            command =>
+                command
+                    .setName(this.name)
+                    .setDescription(this.description)
+                    .addUserOption(user =>
+                        user
+                            .setName(Logs.User)
+                            .setDescription('The user to view')
+                            .setRequired(true)
+                    ),
+            { idHints: [] }
+        );
+    }
+
+    private async renderLogPage(
+        selectedCategory: Logs,
+        index: number,
+        userLog: ModerationLog,
+        cache: CachedPages
+    ) {
+        switch (selectedCategory) {
+            case Logs.User:
+                return this.userPage(userLog, cache);
+            case Logs.Warnings:
+                return this.warningsPage(index, userLog, cache);
+            case Logs.Timeouts:
+                return this.timeoutsPage(index, userLog, cache);
+            case Logs.Bans:
+                return this.bansPage(index, userLog, cache);
+            default:
+                return '';
+        }
+    }
+
+    private async getLogLength(
+        selectedCategory: Logs,
+        userLog: ModerationLog,
+    ) {
+        switch (selectedCategory) {
+            case Logs.Warnings:
+                return userLog.warnings?.length || 1;
```

Comment:
Usually better to use `??` when working with undefined/null.
```suggestion
                return userLog.warnings?.length ?? 1;
```